### PR TITLE
avoid converting types into themselves (clippy::useless_conversion)

### PR DESCRIPTION
**Pull request by matthiaskrgr**

### CODE REVIEW

File: src/librustdoc/core.rs [hunk 1]
```diff
@@ -161,7 +161,7 @@ impl<'tcx> DocContext<'tcx> {
             }
             Entry::Occupied(e) => e.into_mut(),
         };
-        *def_index = DefIndex::from(*def_index + 1);
+        *def_index = *def_index + 1;
```
Comment (jyn514) [inline src/librustdoc/core.rs:164]: Does this compile?
```suggestion
        *def_index += 1;
```

**Review by jyn514 — APPROVED** [review]

File: src/librustdoc/core.rs (hunk 1 above)
Comment (matthiaskrgr) [reply src/librustdoc/core.rs:164]: nope :(
````
error[E0368]: binary assignment operation `+=` cannot be applied to type `DefIndex`
   --> src/librustdoc/core.rs:164:9
    |
164 |         *def_index += 1;
    |         ----------^^^^^
    |         |
    |         cannot use `+=` on type `DefIndex`

````

Comment (jyn514) [reply src/librustdoc/core.rs:164]: Now I'm confused, how does this code work at all? DefIndex implements neither `Deref` nor `Add`.

Comment (matthiaskrgr) [reply src/librustdoc/core.rs:164]: Could be related to the `newtype_index!()` macro that is used when the struct is declared
https://doc.rust-lang.org/nightly/nightly-rustc/src/rustc_span/def_id.rs.html#119
https://doc.rust-lang.org/nightly/nightly-rustc/rustc_index/macro.newtype_index.html

Comment (jyn514) [reply src/librustdoc/core.rs:164]: Ah, ok. It would be nice to add `AddAssign` there. But no need to do it here.

**Pull request merged**